src/picongpu/include/particles/manipulators/IfRelativeGlobalPositionImpl.hpp

```diff
@@ -45,31 +45,55 @@ struct IfRelativeGlobalPositionImpl : private T_Functor
     }
 
     template<typename T_Particle1, typename T_Particle2>
-    DINLINE void operator()(const DataSpace<simDim>& localCellIdx,
+    DINLINE void operator()(const DataSpace<simDim>& localSuperCellOffset,
                             T_Particle1& particle1, T_Particle2& particle2,
                             const bool isParticle1, const bool isParticle2)
     {
         typedef typename SpeciesType::FrameType FrameType;
 
-
-        DataSpace<simDim> myCellPosition = localCellIdx + localDomainOffset;
-
-        float_X relativePosition = float_X(myCellPosition[ParamClass::dimension]) /
-            float_X(globalDomainSize[ParamClass::dimension]);
-
-        const bool inRange=(ParamClass::lowerBound <= relativePosition &&
-            relativePosition < ParamClass::upperBound);
-        const bool particleInRange1 = isParticle1 && inRange;
-        const bool particleInRange2 = isParticle2 && inRange;
-
-        Functor::operator()(localCellIdx,
+        /* offset of the superCell (in cells, without any guards) to the origin of the global domain */
+        const DataSpace<simDim> globalSuperCellOffset = localSuperCellOffset + localDomainOffset;
+        bool particleInRange1 = isParticle1;
+        bool particleInRange2 = isParticle2;
+
+        if( isParticle1 )
+        {
+            particleInRange1 = isParticleInsideRange( particle1, globalSuperCellOffset);
+        }
+        if( isParticle1 )
```

Comment:
As I know this bug has no effect on any currently used example. The binary version of this functor is not used anywhere.